tests/examples/test_check_shapes_examples.py

```diff
@@ -0,0 +1,739 @@
+        return len(x.shape)
+
+    # [dimension_spec_anonymous_variable_rank__star]
+
+    for batch_shape in [(), (2,), (2, 4)]:
+        x = np.ones(batch_shape)
+        rank(x)
+
+
+def test_example__dimension_spec_anonymous_variable_rank__ellipsis() -> None:
+    # [dimension_spec_anonymous_variable_rank__ellipsis]
+
+    @check_shapes(
+        "x: [...]",
+    )
+    def rank(x: AnyNDArray) -> int:
+        return len(x.shape)
+
+    # [dimension_spec_anonymous_variable_rank__ellipsis]
+
+    for batch_shape in [(), (2,), (2, 4)]:
+        x = np.ones(batch_shape)
+        rank(x)
+
+
+def test_example__dimension_spec__scalar() -> None:
+    # [dimension_spec__scalar]
+
+    @check_shapes(
+        "x: [...]",
+        "return: []",
+    )
+    def mean(x: AnyNDArray) -> AnyNDArray:
+        mean: AnyNDArray = np.sum(x) / x.size
+        return mean
+
+    # [dimension_spec__scalar]
+
+    for batch_shape in [(), (2,), (2, 4)]:
+        x = np.ones(batch_shape)
+        mean(x)
+
+
+def test_example__dimension_spec_broadcast() -> None:
+
+    # [dimension_spec_broadcast]
+
+    @check_shapes(
+        "a: [broadcast batch...]",
+        "b: [broadcast batch...]",
+        "return: [batch...]",
+    )
+    def add(a: AnyNDArray, b: AnyNDArray) -> AnyNDArray:
+        return a + b
+
+    # [dimension_spec_broadcast]
+
+    add(np.ones((3, 1)), np.ones((1, 4)))
+
+
+def test_example__bool_spec_argument_ref() -> None:
+    # pylint: disable=unused-argument
+
+    # [bool_spec_argument_ref]
+
+    @check_shapes(
+        "a: [broadcast batch...] if check_a",
+        "b: [broadcast batch...] if check_b",
+        "return: [batch...]",
+    )
+    def add(a: AnyNDArray, b: AnyNDArray, check_a: bool = True, check_b: bool = True) -> AnyNDArray:
+        return a + b
+
+    # [bool_spec_argument_ref]
+
+    add(np.ones((3, 1)), np.ones((1, 4)), check_b=False)
+
+
+def test_example__bool_spec_or() -> None:
+    # pylint: disable=unused-argument
+
+    # [bool_spec_or]
+
+    @check_shapes(
+        "a: [broadcast batch...] if check_all or check_a",
+        "b: [broadcast batch...] if check_all or check_b",
+        "return: [batch...]",
+    )
+    def add(
+        a: AnyNDArray,
+        b: AnyNDArray,
+        check_all: bool = False,
+        check_a: bool = True,
+        check_b: bool = True,
+    ) -> AnyNDArray:
+        return a + b
+
+    # [bool_spec_or]
+
+    add(np.ones((3, 1)), np.ones((1, 4)), check_b=False)
+
+
+def test_example__bool_spec_and() -> None:
+    # pylint: disable=unused-argument
+
+    # [bool_spec_and]
+
+    @check_shapes(
+        "a: [broadcast batch...] if enable_checks and check_a",
+        "b: [broadcast batch...] if enable_checks and check_b",
+        "return: [batch...]",
+    )
+    def add(
+        a: AnyNDArray,
+        b: AnyNDArray,
+        enable_checks: bool = True,
+        check_a: bool = True,
+        check_b: bool = True,
+    ) -> AnyNDArray:
+        return a + b
+
+    # [bool_spec_and]
```

Comment:
Similar to the above, I would include the call in this example.

Reply:
> Similar to the above, I would include the call in this example.

Sure.